modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java

```diff
@@ -336,6 +334,120 @@ public void testValidateTokenScope() throws Exception {
+
+        String userInfoUrl = tenantInfo.getDomain().equalsIgnoreCase("carbon.super") ?
+                OAuth2Constant.USER_INFO_ENDPOINT : OAuth2Constant.TENANT_USER_INFO_ENDPOINT;
+        HttpPost request = new HttpPost(userInfoUrl);
+
+        List<NameValuePair> urlParameters = Collections.singletonList(
+                new BasicNameValuePair("access_token", accessToken)
+        );
+        request.setHeader("User-Agent", OAuth2Constant.USER_AGENT);
+        request.setHeader("Content-Type", "application/x-www-form-urlencoded");
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
+
+        HttpResponse response = client.execute(request);
+
+        String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
+        EntityUtils.consume(response.getEntity());
+        JSONParser parser = new JSONParser();
+        JSONObject jsonResponse = (JSONObject) parser.parse(responseString);
+        if (jsonResponse == null) {
+            throw new Exception("Error occurred while getting the response.");
+        }
+        Assert.assertNotNull(jsonResponse.get("sub"), "sub from introspection endpoint response is null.");
+        Assert.assertNotNull(jsonResponse.get("email"), "sub from introspection endpoint response is null.");
+    }
+
+    @Test(groups = "wso2.is", description = "request user info using POST iwht invalid token",
+            dependsOnMethods = "testUserInfoPostRequest")
+    public void testUserInfoPostWithInvalidToken() throws Exception {
+
+        String userInfoUrl = tenantInfo.getDomain().equalsIgnoreCase("carbon.super") ?
+                OAuth2Constant.USER_INFO_ENDPOINT : OAuth2Constant.TENANT_USER_INFO_ENDPOINT;
+        HttpPost request = new HttpPost(userInfoUrl);
+
+        List<NameValuePair> urlParameters = Collections.singletonList(
+                new BasicNameValuePair("access_token", "invalid_access_token")
+        );
+        request.setHeader("User-Agent", OAuth2Constant.USER_AGENT);
+        request.setHeader("Content-Type", "application/x-www-form-urlencoded");
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
+
+        HttpResponse response = client.execute(request);
+
+        String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
+        EntityUtils.consume(response.getEntity());
+        JSONParser parser = new JSONParser();
+        JSONObject jsonResponse = (JSONObject) parser.parse(responseString);
+        if (jsonResponse == null) {
+            throw new Exception("Error occurred while getting the response.");
+        }
+        Assert.assertEquals(jsonResponse.get("error_description"),
+                "Access token validation failed", "Unexpected error description");
+        Assert.assertEquals(jsonResponse.get("error"), "invalid_token",
+                "Unexpected error message");
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request without openid and invoke introspect",
```

Comment:
is this description correct. we are sending openid specc but its m2m

Reply:
fixed with https://github.com/wso2/product-is/pull/21361/commits/be2f26ad6ce18d03ba9390625c6d1d6ab2962dcd